packages/params/src/forkName.ts

```diff
@@ -20,28 +20,70 @@ export enum ForkSeq {
   deneb = 4,
 }
 
+function exclude<T>(coll: T[], val: T[]): T[] {
+  return coll.filter((f) => !val.includes(f));
+}
+
+export function highestFork<F extends ForkName>(forkNames: F[]): F {
+  let highest = forkNames[0];
+
+  for (const forkName of forkNames) {
+    if (ForkSeq[forkName] > ForkSeq[highest]) {
+      highest = forkName;
+    }
+  }
+
+  return highest;
+}
+
+export function lowestFork<F extends ForkName>(forkNames: F[]): F {
+  let lowest = forkNames[0];
+
+  for (const forkName of forkNames) {
+    if (ForkSeq[forkName] < ForkSeq[lowest]) {
+      lowest = forkName;
+    }
+  }
+
+  return lowest;
+}
+
 export type ForkAll = ForkName;
+export const forkAll = Object.keys(ForkName) as ForkName[];
 
 export type ForkPreLightClient = ForkName.phase0;
 export type ForkLightClient = Exclude<ForkName, ForkPreLightClient>;
+export const forkLightClient = exclude(forkAll, [ForkName.phase0]) as ForkLightClient[];
 export function isForkLightClient(fork: ForkName): fork is ForkLightClient {
   return fork !== ForkName.phase0;
 }
 
 export type ForkPreExecution = ForkPreLightClient | ForkName.altair;
 export type ForkExecution = Exclude<ForkName, ForkPreExecution>;
+export const forkExecution = exclude(forkAll, [ForkName.phase0, ForkName.altair]) as ForkExecution[];
 export function isForkExecution(fork: ForkName): fork is ForkExecution {
   return isForkLightClient(fork) && fork !== ForkName.altair;
 }
 
 export type ForkPreWithdrawals = ForkPreExecution | ForkName.bellatrix;
 export type ForkWithdrawals = Exclude<ForkName, ForkPreWithdrawals>;
+export const forkWithdrawals = exclude(forkAll, [
+  ForkName.phase0,
+  ForkName.altair,
+  ForkName.bellatrix,
+]) as ForkWithdrawals[];
 export function isForkWithdrawals(fork: ForkName): fork is ForkWithdrawals {
   return isForkExecution(fork) && fork !== ForkName.bellatrix;
 }
 
 export type ForkPreBlobs = ForkPreWithdrawals | ForkName.capella;
 export type ForkBlobs = Exclude<ForkName, ForkPreBlobs>;
+export const forkBlobs = exclude(forkAll, [
```

Comment:
can't we just do the follow here:

```ts
export const forkBlobs = exclude(forkAll, forkWithdrawals)
```

same for the others above

Reply:
I prefer to be more explicit. Otherwise just to aware what is `forkBlobs` one have to go up and check what is `forkWithdrawals` and so on.

Reply:
but we do the same for types, e.g.
```ts
export type ForkPreBlobs = ForkPreWithdrawals | ForkName.capella;
```

Reply:
Those are types, if you hover those will give you exact union of each fork names. But for values you will just see `ForkExecution[]` or `ForkName[]`, so it's more readable when we have explicit values here.

Reply:
I am more thinking about this in terms of what's the less error prone and simple approach if we add a new fork

Reply:
There is not much to think, more readable code is less error prone.

Reply:
But we are using a more error prone approach right now

```ts
export const forkBlobs = exclude(forkAll, [
  ForkName.phase0,
  ForkName.altair,
  // ForkName.bellatrix, <-- this can be removed without any complaints
  ForkName.capella,
]) as ForkBlobs[];
```
Readability here is subjective, for me a simple oneliner that follows same pattern as types is more readable but it's clearly less error prone, especially if you add a new fork (also less work to add a new fork).

Reply:
- https://github.com/ChainSafe/lodestar/pull/6916 addresses issue in my previous comment by removing type casts